### PR TITLE
finish week1 progress

### PR DESCRIPTION
**Pull request by eunice012716**
### Todo
- [x] ch2
- [x] ch3

### 道歉文
真的很抱歉ＱＡＱ因為我windows系統的電腦的ubuntu壞掉了所以也沒辦法連server解決我linter的問題
拖到現在才給進度真的很不好意思
ch4 ch5也完成了 如果這個ＰＲ過的話我再上傳ch4 ch5的進度
真的很不好意思

### CODE REVIEW

File: eunice012716/Week1/ch2/2.2/exercise1.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        for _ in range(0, 40):
+            id_ = str(get_random_id(10000, 20000, 0.3)) + ","
+            gen_ = str(random.choice(gender)) + ","
+            age_ = str(get_random_age(24, 55, 0.3)) + ","
+            d_ = str(random.choice(department)) + "\n"
+            data_file.write(id_ + gen_ + age_ + d_)
+    data = pd.read_csv(data_path)
+    print(data)
+
+    print()
```
Comment (jamfly) [inline eunice012716/Week1/ch2/2.2/exercise1.py:48]: 可以改成 `print(data, end="\n\n")` 或是中間不要有兩個空格

File: eunice012716/Week1/ch2/2.4/exercise1.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    display.set_matplotlib_formats("svg")
+
+
+def set_figsize(figsize=(3.5, 2.5)):  # @save
+    """Set the figure size for matplotlib."""
+    use_svg_display()
+    d2l.plt.rcParams["figure.figsize"] = figsize
+
+
+# @save
```
Comment (jamfly) [inline eunice012716/Week1/ch2/2.4/exercise1.py:25]: 這個要拿掉

File: eunice012716/Week1/ch2/2.4/exercise1.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+def set_axes(axes, xlabel, ylabel, xlim, ylim, xscale, yscale, legend):
+    """Set the axes for matplotlib."""
+    axes.set_xlabel(xlabel)
+    axes.set_ylabel(ylabel)
+    axes.set_xscale(xscale)
+    axes.set_yscale(yscale)
+    axes.set_xlim(xlim)
+    axes.set_ylim(ylim)
+    if legend:
```
Comment (jamfly) [inline eunice012716/Week1/ch2/2.4/exercise1.py:34]: 跟前面 miss 的問題一樣

File: eunice012716/Week1/ch3/3.2/exercise2.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+if __name__ == "__main__":
+    true_w = torch.Tensor([3.3])
+    true_b = 0
+    currents, voltages = synthetic_data(true_w, true_b, 1000)
+
+    w = torch.normal(0, 0.01, size=(1,), requires_grad=True)
+    b = torch.normal(0, 0.01, size=(1,), requires_grad=True)
+    net = linreg
+    loss = squared_loss
```
Comment (jamfly) [inline eunice012716/Week1/ch3/3.2/exercise2.py:56]: loss 會讓人誤會是已經計算好的值，這邊不需要再賦值，原本的名字就可以知道他是啥

**Review by jamfly — CHANGES_REQUESTED** [review]

**Comment by jamfly** [comment]
ch4. ch5 可以同時推上來，基本上是不會有影響，只是你必須 rebase/merge main 這條 branch，在你 merge 其中一條 pr 之後

**Review by jamfly — APPROVED** [review]
